middleware/compress.go

```diff
@@ -0,0 +1,186 @@
+package middleware
+
+import (
+	"compress/flate"
+	"compress/gzip"
```

Comment:
I think good idea. You could take a Compressor or something, or use a default one if its not passed